### PR TITLE
Modernize dependencies

### PR DESCRIPTION
**Pull request by jvansanten**
This PR addresses compatibility issues with modern versions of PyTorch, LightGBM, and the Python standard library that have made it increasingly difficult to install dependencies with recent versions of Python.

* Remove version cap on `lightgbm` for compatibility with LightGBM >= 4.
* Remove the `verbose=True` argument from the learning rate scheduler initialization for compatibility with PyTorch>=2.6
* Load models with `weights_only=False` for compatibility with PyTorch>=2.6
* Switch from `pkg_resources` to `importlib.resources` for compatibility with the Python 3.12 standard library.

### CODE REVIEW

**Review by hombit — APPROVED** [review]
@jvansanten thank you, looks good! I'll release a new version 1.4.3 ASAP

**Comment by hombit** [comment]
@jvansanten this would also need python 3.9+, I'll update `setup.cfg` for that

**Comment by hombit** [comment]
@jvansanten v1.4.3 was released:
https://pypi.org/project/astro-parsnip/1.4.3/

Our release workflow doesn't work, so I released it manually. Please let me know if I broke anything!
